### PR TITLE
Mailet 149 raw ical

### PR DESCRIPTION
**Pull request by rouazana**
Please review last commit

### CODE REVIEW

File: mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageIdMapper.java [hunk 1]
```diff
@@ -289,7 +289,7 @@ public void delete(MessageId messageId) {
         return imapUidDAO.updateMetadata(newComposedId, oldComposedId.getModSeq())
             .thenCompose(updateSuccess -> Optional.of(updateSuccess)
                 .filter(b -> b)
-                .map(any -> messageIdDAO.updateMetadata(newComposedId).thenApply(v -> updateSuccess))
+                .map((Boolean any) -> messageIdDAO.updateMetadata(newComposedId).thenApply(v -> updateSuccess))
```
Comment (strokyl) [inline mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageIdMapper.java]: Which one is going to be merge first ;-) https://github.com/linagora/james-project/pull/575/commits/658d5cfcae3362b1de754f82c863bb5c30025860

File: server/mailet/integration-testing/src/test/java/org/apache/james/transport/mailets/amqp/AmqpRule.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+    public AmqpRule(SwarmGenericContainer rabbitMqContainer, String exchangeName, String routingKey) {
+        this.rabbitMqContainer = rabbitMqContainer;
+        this.exchangeName = exchangeName;
+        this.routingKey = routingKey;
+    }
+
+    @Override
+    protected void before() throws Throwable {
+        @SuppressWarnings("deprecation")
```
Comment (strokyl) [inline server/mailet/integration-testing/src/test/java/org/apache/james/transport/mailets/amqp/AmqpRule.java]: Firstly @suppressWarnings does not compile on my eclipse (I have to put it at the method level). I guess it's again eclipse that suck.
Secondly, I have the exact same ugly line here: https://github.com/linagora/james-project/pull/559/files#diff-450812b86702bd99720bdc0a18b4703eR75.
I am afraid the day where we will figure out how to do that with the "new api" we will have a lot of line to modify.
If it's not too much too ask, could you refactor that.

File: server/mailet/integration-testing/src/test/java/org/apache/james/transport/mailets/amqp/AmqpRule.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+            .atMost(30, TimeUnit.SECONDS)
+            .with()
+            .pollInterval(10, TimeUnit.MILLISECONDS)
+            .until(() -> isReady(factory));
+    }
+
+    private boolean isReady(ConnectionFactory factory) {
+        try (Connection connection = factory.newConnection()) {
+            return true;
+        } catch (IOException e) {
```
Comment (strokyl) [inline server/mailet/integration-testing/src/test/java/org/apache/james/transport/mailets/amqp/AmqpRule.java]: `catch (IOException|TimeoutException e)`

File: server/mailet/integration-testing/src/test/java/org/apache/james/transport/mailets/amqp/AmqpRule.java (hunk 2 above)
Comment (chibenwa) [inline server/mailet/integration-testing/src/test/java/org/apache/james/transport/mailets/amqp/AmqpRule.java]: You are right about getting it into swarm container

For your @SuppressWarnings, I guess we can move it on an extractes method.

File: server/mailet/integration-testing/src/test/resources/yahooInvitation.eml [hunk 4]
```diff
@@ -0,0 +1,193 @@
+Return-Path: <do-not-reply@yahoo.com>
```
Comment (chibenwa) [inline server/mailet/integration-testing/src/test/resources/yahooInvitation.eml]: Anonimize

**Pull request closed without merging**